lib/gdpr/gateway/session.rb

```diff
@@ -6,10 +6,15 @@ def delete_sessions
 
     logger.info('Starting daily session deletion')
 
-    dataset = DB['DELETE FROM sessions WHERE start < DATE_SUB(DATE(NOW()), INTERVAL 32 DAY)']
-    rows_affected = dataset.delete
+    rows = Session.where(Sequel.lit('start < DATE_SUB(DATE(NOW()), INTERVAL 32 DAY')).count
 
-    logger.info("Finished daily session deletion, #{rows_affected} rows affected")
+    i = 0
+    while i < rows
+      Session.where(Sequel.lit('start < DATE_SUB(DATE(NOW()), INTERVAL 32 DAY')).limit(100, i).delete
```

Comment:
I am not sure how Ruby Session/Sequel works, but many databases/drivers provide you will ability to get a cursor and paginate through it. That would eliminate the need for getting a count first and may provide more efficient queries.

Reply:
I wonder, if you're deleting and it's ACID, then you are going to get a different batch of 100 items every time, so the offset may not be needed (and may even break from overflow)

Reply:
Fair point, though I don't think we need to make it acid compliant.  The intention of the query is to nuke old sessions greater than 32 days, there will be a slight drift in calculated time between each iteration of the while loop when each of the separate queries are made, but that will be small and addressed upon a subsequent run.   Even if it is a different set of 100 per iteration, it'll still end up nuking the oldest sessions.  Is this what you mean @phantas ?

Reply:
I think we are in the same page, but just to clarify: the issue is in every loop iteration you do a transaction. So those 100 records won't exist in the next loop iteration but you are still offsetting by 100. You can instead just delete the first 100 records and, for extra points, stop when the # of records returned is zero (so you don't even need a count). That's the usual pattern, but there's nothing wrong with using the count instead.

Reply:
Thanks @phantas, now refactored!